.cirrus.yml

```diff
@@ -1,51 +1,49 @@
-BUILD_TEST_TASK_TEMPLATE: &BUILD_TEST_TASK_TEMPLATE
+TEST_TEMPLATE: &TEST_TEMPLATE
   arch_check_script:
     - uname -am
   test_script:
     - python --version
-    - python -m pip install --upgrade pip
-    - python -m pip install -r requirements_dev.txt
-    - python -m flake8
-    - python -m pydocstyle pact
-    - python -m tox -e test
-    # - make examples
+      # TODO: Fix lints before enabling
+    - echo hatch run lint
+      # TODO: Implement the examples to work in hatch
+    - echo hatch run example
+    - hatch run test
 
-linux_arm64_task: 
+linux_arm64_task:
   env:
+    PATH: ${HOME}/.local/bin:${PATH}
     matrix:
-      # - IMAGE: python:3.6-slim  # This works locally, with cirrus run, but fails in CI
-      - IMAGE: python:3.7-slim
       - IMAGE: python:3.8-slim
       - IMAGE: python:3.9-slim
       - IMAGE: python:3.10-slim
+      - IMAGE: python:3.11-slim
   arm_container:
     image: $IMAGE
   install_script:
-    - apt update --yes && apt install --yes gcc make
-  << : *BUILD_TEST_TASK_TEMPLATE
-
+    - apt update --yes
+    - apt install --yes gcc make
+    - python -m pip install --upgrade pip pipx
+    - pipx install hatch
+  <<: *TEST_TEMPLATE
 
 macosx_arm64_task:
   macos_instance:
     image: ghcr.io/cirruslabs/macos-ventura-base:latest
   env:
-    PATH: ${HOME}/.pyenv/shims:${PATH}
+    PATH: ${HOME}/.local/bin:${HOME}/.pyenv/shims:${PATH}
     matrix:
-      - PYTHON: 3.6
-      - PYTHON: 3.7
       - PYTHON: 3.8
       - PYTHON: 3.9
       - PYTHON: 3.10
+      - PYTHON: 3.11
   install_script:
-    # Per the pyenv homebrew recommendations.
-    # https://github.com/pyenv/pyenv/wiki#suggested-build-environment
-    # - xcode-select --install  # Unnecessary on Cirrus
-    - brew update    
-    # - brew install openssl readline sqlite3 xz zlib 
+    - brew update
     - brew install pyenv
     - pyenv install ${PYTHON}
     - pyenv global ${PYTHON}
     - pyenv rehash
-  ## To install rosetta
-    # - softwareupdate --install-rosetta --agree-to-license
-  << : *BUILD_TEST_TASK_TEMPLATE
+    - python -m pip install --upgrade pip pipx
+    - pyenv rehash
+    - pipx install hatch
```

Comment:
do we always want to run against the latest version of hatch, or do we want to pin this so it could be picked up via say a dependabot config (or similar) to ensure we can test version updates and ensure consistency.

Reply:
as a note, my first experience was that I needed to install hatch, which is where I came back to the CI scripts to see how we are installing it :)

```
pact-python on  feat/pyproject@origin:feat/pyproject via 🐳 desktop-linux via 🐍 v3.11.4 on ☁️  (eu-west-2) 
🕙15:02:03 ❯ make                                                

  clean      to clear build and distribution directories
  examples   to run the example end to end tests (consumer, fastapi, flask, messaging)
  consumer   to run the example consumer tests
  fastapi    to run the example FastApi provider tests
  flask      to run the example Flask provider tests
  messaging  to run the example messaging e2e tests
  package    to create a distribution package in /dist/
  release    to perform a release build, including deps, test, and package targets
  test       to run all tests


pact-python on  feat/pyproject@origin:feat/pyproject via 🐳 desktop-linux via 🐍 v3.11.4 on ☁️  (eu-west-2) 
🕙15:02:40 ❯ make clean
hatch clean
make: hatch: No such file or directory
make: *** [clean] Error 1

pact-python on  feat/pyproject@origin:feat/pyproject via 🐳 desktop-linux via 🐍 v3.11.4 on ☁️  (eu-west-2) 
🕙15:02:44 [🔴 USAGE] ❯ make package
hatch build
make: hatch: No such file or directory
make: *** [package] Error 1

pact-python on  feat/pyproject@origin:feat/pyproject via 🐳 desktop-linux via 🐍 v3.11.4 on ☁️  (eu-west-2) 
🕙15:02:51 [🔴 USAGE] ❯ make test   
hatch run all
make: hatch: No such file or directory
make: *** [test] Error 1
```

Reply:
Good question. I think installing the latest version of hatch should generally not be an issue. Hatch is not an actually dependency of `pact-python`, and instead is just a CLI interface and a build backend. I would be very surprised if regular updates to hatch broke this.

Of course, should hatch come out with a completely overhaul in `v2`, then we might need to pin it then, but I'm happy to leave this as future problem.

In the same vein, I'm of two minds as to whether we should update system dependencies (`apt` and `brew`) as part of the workflow. Personally, I typically lean towards faster CI/CD builds and therefore use whatever packages are installed in the image, and then trust the CI/CD provider to regularly update these packages.

Lastly, would you like me to add a `make` command to install hatch? Or not really worth it?

Reply:
>Of course, should hatch come out with a completely overhaul in v2, then we might need to pin it then, but I'm happy to leave this as future problem.

Yeah I'm okay with that, the CI will catch any changes, and alert, and we can action accordingly

>In the same vein, I'm of two minds as to whether we should update system dependencies (apt and brew) as part of the workflow. Personally, I typically lean towards faster CI/CD builds and therefore use whatever packages are installed in the image, and then trust the CI/CD provider to regularly update these packages.

You get a more repeatable build by not pulling in new updates each version, I've always just updated as a matter of course, going from practises used in Docker files, but also find it incredibly annoying when building things locally that link dynamically, and the dep has been updated with a new named shared lib, causing my built programs to fail, until they are rebuilt. Also why I like statically compiled executables, that will always work, even if they are chonkier.

> Lastly, would you like me to add a make command to install hatch? Or not really worth it?

could do, if we did I would favour it being installed with python, so it saves x platform package manager concerns

if we don't, just have it as a pre-req for development somewhere in the docs.

you could have a make doctor command, which checks if the hatch executable is available and alerts people on how to install it